### PR TITLE
[HB-9335] Drafts documentation

### PR DESCRIPTION
**Pull request by elagulsen**
Not merging until the feature is ready to launch ⚠️

~TODO: update the permissions section now that we've changed that behavior~ Done ✅

Explains the flow for creating a draft, permissions, validation, and applying a draft. Let me know if there's anything else I missed that would be helpful for a user to know.

### CODE REVIEW

**Comment by nstokoe** [comment]
Is there a way we can release this but not have it as a link in the nav or have it be searchable. If we can do that, then we link to it in the code before release day.

**Comment by elagulsen** [comment]
@nstokoe if I add `"display": "hidden"` in `_meta.json` it does hide the page from the nav, but not from search. So it's still possible for someone to see that page by accident if they search `draft` or `validation` or something

**Comment by nstokoe** [comment]
Ok, I did some googling and it seems there is not an easy way to de-list content from the search without getting into the internals of `nextra` and `flexsearch`. We can leave this until release day. We just have to remember to update the links in the app.

File: pages/docs/project-management/drafts.mdx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Drafts live in the [Drafts page](https://hashboard.com/app/p/drafts) within a project. 
+
+To create a draft, click the `New draft` button. You'll be prompted for a name and description, which can be changed later.
+
+Once you've created a draft, you can enter the draft by clicking on the `Start drafting` button. 
+
+While in a draft, you should see an orange `Drafting` button at the top of the page to indicate that you're in a draft. 
+This means you're safe to interact with the project and make changes to resources, knowing that they will not be applied to your project until you're ready.
+
+To exit the draft, click the `X` button on the orange `Drafting` button at the top of the page, or navigate back to the Drafts page.
```
Comment (nstokoe) [inline pages/docs/project-management/drafts.mdx:20]: Could mention there are two buttons here? X leaves you on the current resource just not in draft and the "Drafting" button takes you back to the draft details page.

File: pages/docs/project-management/drafts.mdx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+There are no special permissions required for users to create, view, or apply drafts. 
+However, users have the same permissions within draft mode as they do when interacting with other project resources. 
+
+For example, if a user does not have permission to create a model, they will still not be able to create a model in a draft.
+
+## Validating drafts
+
+When developing a draft, it's possible that the draft may enter an invalid state. 
+
+Similar to a merge conflict, this can happen if changes are made to the project that are incompatible with the changes in the draft.
```
Comment (nstokoe) [inline pages/docs/project-management/drafts.mdx:33]: Maybe say "... to a merge conflict in git..." people that aren't aware of git at least have some context clues to search this out on their own.

**Review by nstokoe — APPROVED** [review]
Overall I think these are a great starting point, but will wait for @vndrewlee to chime in as well.

File: pages/docs/project-management/drafts.mdx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+---
+description: Hashboard drafts allow you to stage changes to your project safely and apply them in a single operation when you're ready. 
+---
+
+# Drafts
+
+Drafts are a way to stage changes to your Hashboard project safely.
```
Comment (vndrewlee) [inline pages/docs/project-management/drafts.mdx:7]: maybe

```suggestion
Drafts are a way to test or propose changes to your Hashboard project safely. 
```

**Review by vndrewlee — APPROVED** [review]
lgtm

I would probably add a bit describing that drafts are public and shared across the project. Also mentioning that you can share links to your draft internally for review.

**Comment by nstokoe** [comment]
Maybe we should add a section on what things are not part of drafts, most importantly:
- data connections
- project settings
- users
- roles

**Pull request merged**